README.md

```diff
@@ -16,43 +16,50 @@ Installation requires the following steps:
+1. Click the `Grant Permissions` button and confirm by clicking `Yes`. **Note:** only AD tenant admin can grant permissions to an Azure AD application.
+1. On the `Settings` panel of the application, select `Keys`.
+1. Enter key `Description` and set `Duration` to `Never expires` then click `Save`.
+**Note:** please save the key value, it is needed later during template deployment.
+1. Get the `Service Principal Id` associated with the application by navigating back to the `Registered App` blade, click on the link under `Managed application in local directory` then click `Properties`. The `Service Principal Id`
+is labeled as `Object ID` on the properties page.
+**Caution:** This is not the same `Object ID` found under the `Registered app` view or under the `Settings`.
 
 ## Create an Alert Logic Access Key
 
 From the Bash command line in [Azure Cloud Shell](https://docs.microsoft.com/en-us/azure/cloud-shell/quickstart) run the following commands, where `<username>` is your Alert Logic user and `<password>` is your Alert Logic password:
+
 ```
 export AL_USERNAME='<username>'
 auth=$(curl -X POST -s -u $AL_USERNAME https://api.global-services.global.alertlogic.com/aims/v1/authenticate); export AL_ACCOUNT_ID=$(echo $auth | jq -r '.authentication.account.id'); export AL_USER_ID=$(echo $auth | jq -r '.authentication.user.id'); export AL_TOKEN=$(echo $auth | jq -r '.authentication.token'); if [ -z $AL_TOKEN ]; then echo "Authentication failure"; else roles=$(curl -s -X GET -H "x-aims-auth-token: $AL_TOKEN" https://api.global-services.global.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/roles | jq -r '.roles[].name'); if [ "$roles" != "Administrator" ]; then echo "The $AL_USERNAME doesn’t have Administrator role. Assigned role is '$roles'"; else curl -s -X POST -H "x-aims-auth-token: $AL_TOKEN" https://api.global-services.global.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/access_keys | jq .; fi; fi; unset AL_USERNAME;
 ```
+
+For accounts where MFA is enabled:
+```
+export AL_USERNAME='<username>'
+auth=$(curl -sX POST -d '{"mfa_code": "<mfa_code_here>" }' -u $AL_USERNAME https://api.global-services.global.alertlogic.com/aims/v1/authenticate); export AL_ACCOUNT_ID=$(echo $auth | jq -r '.authentication.account.id'); export AL_USER_ID=$(echo $auth | jq -r '.authentication.user.id'); export AL_TOKEN=$(echo $auth | jq -r '.authentication.token'); if [ -z $AL_TOKEN ]; then echo "Authentication failure"; else roles=$(curl -s -X GET -H "x-aims-auth-token: $AL_TOKEN" https://api.global-services.global.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/roles | jq -r '.roles[].name'); if [ "$roles" != "Administrator" ]; then echo "The $AL_USERNAME doesn’t have Administrator role. Assigned role is '$roles'"; else curl -s -X POST -H "x-aims-auth-token: $AL_TOKEN" https://api.global-services.global.alertlogic.com/aims/v1/$AL_ACCOUNT_ID/users/$AL_USER_ID/access_keys | jq .; fi; fi; unset AL_USERNAME;
+```
+
```

Comment:
curl -s will hide error message, no? maybe it's better to drop that switch?

Reply:
Actually, just -S is redundant (I think); it's either -sS (silent, but show errors), -s (silent, don't show errors), or no "s" option (show progress and errors); still, the next check for AL_TOKEN will work anyway.